src/TheThingsNetworkServer.js

```diff
@@ -0,0 +1,196 @@
+import Server from './Server.js';
+
+import getBytesFromHex from 'jooby-codec/utils/getBytesFromHex.js';
+import getBase64FromBytes from 'jooby-codec/utils/getBase64FromBytes.js';
+
+import {HIGHEST} from './constants/ttn/schedulePriorityTypes.js';
+
+
+const NS_RELAY_NAMESPACE = 'ttn';
+
+const defaultHeaders = {
+    Accept: 'application/json',
+    'Content-Type': 'application/json'
+};
+
+const getHeaders = token => ({
+    method: 'GET',
+    headers: {
+        ...defaultHeaders,
+        Authorization: `Bearer ${token}`
+    }
+});
+
+const getDevice = async ( deviceId, {apiUrl, token, appId} ) => {
+    const response = await fetch(
+        new URL(`${apiUrl}/applications/${appId}/devices/${deviceId}`),
+        {
+            method: 'GET',
+            headers: getHeaders(token)
+        }
+    );
+
+    if ( !response.ok ) {
+        throw new Error('Failed to get device info.');
+    }
+
+    return response.json();
+};
+
+const getDevices = async ( {apiUrl, token, appId} ) => {
+    const response = await fetch(
+        new URL(`${apiUrl}/applications/${appId}/devices`),
+        {
+            method: 'GET',
+            headers: getHeaders(token)
+        }
+    );
+
+    if ( !response.ok ) {
+        throw new Error('Failed to get application devices.');
+    }
+
+    return (await response.json()).end_devices;
+};
+
+
+const sendData = async ( deviceId, data, {nsRelayUrl} ) => {
+    const response = await fetch(
+        new URL(`/${NS_RELAY_NAMESPACE}/${deviceId}/messages`, nsRelayUrl),
+        {
+            method: 'POST',
+            headers: {
+                ...defaultHeaders
+            },
+            body: JSON.stringify({data})
+        }
+    );
+
+    if ( !response.ok ) {
+        throw new Error('Failed to send device payload.');
+    }
+
+    return response.json();
+};
+
+const sendMessage = async ( deviceId, data, {apiUrl, token, appId} ) => {
+    const options = {
```

Comment:
would be nice to make all such functions similar

either use `const options` (**preferably** as already used in chirpstack) or put options inline in fetch call (as in other such methods in this file)

Reply:
rework requests in ChirpStack according to TTN